spec/data-types-and-values.html

```diff
@@ -7,12 +7,63 @@ <h1>ECMAScript Language Types</h1>
 
     <emu-clause id="sec-ecmascript-language-types-record-type">
       <h1>The Record Type</h1>
-      <p>TBD</p>
+      <p>
+        The Record type is the set of all finite mappings from Strings to ECMAScript primitive values including Record and Tuple. Each record value holds an associated [[Fields]] List value which is a list of pairs of the form { [[Key]], [[Value]] } where the [[Key]] is a String and [[Value]] is any primitive value. Entries of [[Fields]] are sorted by [[Key]] in code unit order. The [[Fields]] List and its entries are never modified.
+     </p>
+
+      <emu-clause id="sec-makerecordvalue" aoid="MakeRecordValue">
+        <h1>MakeRecordValue ( _entries_ )</h1>
+        <p>This abstract operation MakeRecordValue takes a _entries_ argument. It is used to specify the creation of new Record value types. It performs the following steps when called:</p>
+        <emu-alg>
+            1. Assert: _entries_ is a List.
+            1. Let _uniqueEntries_ be an empty Set.
+            1. For each _kv_ in _entries_:
+              1. Assert: _kv_.[[Key]] is a String.
+              1. Assert: _kv_.[[Value]] is a primitive type.
+              1. For each _existing_ in _uniqueEntries_:
+                1. If SameValue(_existing_.[[Key]], _kv_.[[Key]]
+                  1. Remove _existing_ from _uniqueEntries_
+                  1. Break
+              1. Add _kv_ to _uniqueEntries_
+            1. Let _sortedEntries_ be a new List containing the same values as the list _entries_ where the values are ordered as if an Array of the same values had been sorted using `Array.prototype.sort` using <emu-xref href="#sec-recordentrycompare">RecordEntryCompare</emu-xref> as _comparefn_.
+            1. Let _R_ be an empty Record
+            1. Set _R_.[[Field]] to _sortedEntries_

```

Comment:
To avoid making it look like a primitive is being mutated (which wouldn't make sense), this could be reworded similarly to [the creation of Symbols](https://tc39.es/ecma262/#sec-symbol-description) as,

Let _R_ be a new Record value whose [[Fields]] value is _sortedEntries_